### PR TITLE
Fix GCC Wreorder Warnings 

### PR DESCRIPTION
**Pull request by DhairyaBahl**
Fix GCC Wreorder warning in the line 209. Do i have to change it elsewhere. Kindly guide 

Thanks :)

### CODE REVIEW

File: synfig-studio/src/gui/workarea.h [hunk 1]
```diff
@@ -206,7 +206,7 @@ class WorkArea : public Gtk::Grid, public Duckmatic
 	int dirty_trap_queued;
 
 	// This flag is set if onion skin is visible
-	bool onion_skin;
+	bool onion_skin;	‘studio::WorkArea::onion_skin’ will be initialized after [-Wreorder]
```
Comment (Keyikedalube) [inline synfig-studio/src/gui/workarea.h:209]: The warning messages were about the object's member initialization order...
You should have at least basic knowledge of OOP to understand and resolve the issue

**Comment by DhairyaBahl** [comment]
@Keyikedalube Yes i am able to understand what is happening here. But i don't know about Wreorder or what is it .

File: synfig-studio/src/gui/workarea.h (hunk 1 above)
Comment (DhairyaBahl) [reply synfig-studio/src/gui/workarea.h:209]: Kindly tell me what is and how to use wreorder 

Thanks :)

**Comment by Keyikedalube** [comment]
Wreorder message is triggered when object member initialization is not in order with the class declaration. So in C++ you have class declaration in `*.h` file and class definition in `*.cpp` file. What you are looking to fix for is the order of the class members to remain the same
```
// declaration.h
class A {
    int a;
    float b;
}
```
```
// definition.cpp
A::A():
    a(0),
    b(0.0) // <--- notice the initialization order
           // a was declared first so it's initialized first
           // and then b is initialized
{
}
```

What we have in synfig is something like this:
```
// definition.cpp
A::A():
    b(0.0),
    a(0) // <--- notice the initialization order
           // b went first instead of a
           // that's why Wreorder warning is triggered
{
}
```
`a` should be at top and then `b` below.

Read this section [Using Initialization Lists to Initialize Fields](https://www.cprogramming.com/tutorial/initialization-lists-c++.html) to better understand the topic

**Comment by ice0** [comment]
It warns you because if this:
```
// declaration.h
class A {
    int a;
    int b;
}
```

```
// definition.cpp
A::A():
    b(5),
    a(b) // <--- You think were will be 5, because you initialized it with b, 
         //      but b will be initialized AFTER a (as you declared)
{
}
```

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube Thanks for putting this much effort for me. I will study these and then review and improve my changes .

Thanks :)

**Comment by DhairyaBahl** [comment]
@rodolforg Okay, If i am not wrong, I dont have to made changes in .h file rather than i have to make them in .cpp file !!
correct me if i am wrong. 

Thanks :)

**Comment by DhairyaBahl** [comment]
@rodolforg @ice0 @Keyikedalube 

highlight_active_bone(false) was getting declared after onion_skin and I shifted it to above show_grid(false) that must be giving the compile time error of Wreorder warnings 

Kindly look after my PR.

Thanks :)

**Comment by Keyikedalube** [comment]
You fixed the issue with this commit https://github.com/synfig/synfig/pull/1750/commits/d0991942ee92b96bd9abbb5de30554ce96a3ca1c :)

But your PR is dirty with other commits and an unneeded merge. You should only rebase when you wish to update your working branch. Also, rebasing is never really needed unless someone's PR was merged first && he modified the same file that you are working on.

Upto project maintainers to decide what's next.
Good job fixing the issue :+1:

**Comment by DhairyaBahl** [comment]
@Keyikedalube Okay Thanks for the advice. Actually @AYESDIE Yesterday told me to rebase properly so that PRs will be approved quickly. That's why I thought i should do the same with every branch i have. Sorry :)

**Comment by DhairyaBahl** [comment]
@Keyikedalube Thanks a lot for giving me that much of hints. Without them I wouldn't be able to solve the issue.

Thanks :)

**Comment by AYESDIE** [comment]
@Keyikedalube There were some issues that were there with the Appveyor CI (some key related issue) which were fixed later in master so the PRs needed a rebase for the checks to be passed. 

And I don't think there's need for 6 commits for a small change, so this one does need cleaning in my opinion. Rebasing and squasing commits will help (^_^)

**Comment by DhairyaBahl** [comment]
@Keyikedalube Actually I was confused about it too. 
When I undo my doing in any file it also counts it as another commit. IDK why ?

Anyways Thank You :)

**Comment by Keyikedalube** [comment]
@DhairyaBahl Happens to everyone :)
Now you know how git works. Every changes you make && commit are recorded.

@AYESDIE I remember rebasing and pushing updates would end up looking like this "*user force-pushed ...*" but his update says "*merge*" so he could have done it some other way. IDK how.

And yes he needs to squash those commits together.

But @DhairyaBahl there's more warning messages about `-Wreorder`
Go trach those and resolve 'em :)
Then squash later not now OK

**Comment by ice0** [comment]
It would also be nice if you give good name for your PR. For example "Fix gcc wreorder warning" or something like that.

**Comment by DhairyaBahl** [comment]
@Keyikedalube Actually I reverse merged the commits from synfig/master to my own dhairyabahl/master that's why it shows merged.

I will now fix the other wreorder warnings.

Thanks for support :)

**Comment by DhairyaBahl** [comment]
@Keyikedalube I tried to find out other wreorder issues under the workarea class but everything seems to be in proper position. 
are the other wreorder warnings present under the same class or under a diff. class ?

Thanks :)

**Comment by Keyikedalube** [comment]
@DhairyaBahl I guess you are building synfig without cleaning. That's why you are not seeing other -Wreorder messages.

You must clean your previous build files and build from start again to see all the warning messages pop up on your console or IDE build log.

The thing about the buildsystem is it is smart to not compile unmodified files all over again every time you build. When you modified the workarea file, make (or whatever build generator you used) compiled only that file. All the other cpp source files weren't needed to be compiled again. Saving you build time...

**Comment by Keyikedalube** [comment]
> are the other wreorder warnings present under the same class or under a diff. class ?

different class

**Comment by DhairyaBahl** [comment]
@Keyikedalube Okay thanks for guidance. I will find those errors and rectify them ASAP. 

Thanks :)

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube I have made changes for all the wreorder warnings that were shown in the console. Kindly look into my PR. 

Thanks :)

**Comment by ice0** [comment]
@DhairyaBahl please fix indents. Synfig uses tabs instead of spaces. You can see difference on this tab: https://github.com/synfig/synfig/pull/1750/files

**Comment by DhairyaBahl** [comment]
@ice0 Ok. I am on it .

Thanks :+1:

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube I rectified the unnecessary use of spaces and properly used tabs there. Can anyone tell me now how to convert 4 commits into 1 ??
Is squashing is the term required here ??

Thanks :)

**Comment by Keyikedalube** [comment]
> Is squashing is the term required here ??

Yes

**Comment by Keyikedalube** [comment]
@DhairyaBahl But wait for ice0 to tell you when. Don't squash now.
He might need to review it first ok

**Comment by DhairyaBahl** [comment]
@Keyikedalube sorry, I already did that 😅

**Comment by Keyikedalube** [comment]
@DhairyaBahl Doesn't matter :) You can still squash even after making review changes.

**Comment by DhairyaBahl** [comment]
@Keyikedalube got it.  Thanks :)

**Review by ice0 — APPROVED** [review]

**Comment by DhairyaBahl** [comment]
@ice0 @Keyikedalube I just completed my 4 valid hacktoberfest PRs. what should I do next?? Take another issue ??

Thanks for help ;)

**Comment by ice0** [comment]
@DhairyaBahl of course! Any help is appreciated!
Even if you just check if an some IDE can open and build Synfig, it also helps a lot, because it all takes time and attention.

**Comment by ice0** [comment]
Merged. Thank you!

**Comment by DhairyaBahl** [comment]
@Keyikedalube @ice0 Please guide me what steps should I take if I want to get in GSOC with this organisation ??
Should I keep contributing to these issues ? OR should i work on some projects for the time being ??

@AYESDIE what should I do ?? I mean I saw ur name in the Ex-Gsocer ? Bro Kindly guide me !!

Thanks :)

**Comment by ice0** [comment]
Just continue to learn the code and make contributions.
Also you can check last year GSoC ideas list here (https://synfig-docs-dev.readthedocs.io/en/latest/gsoc/2020/ideas.html)

We haven't decided on ideas for next year, so maybe you can come up with your own idea.

**Comment by DhairyaBahl** [comment]
@ice0 I will keep contributing in this and will also read that linked documentation just keep guiding me like you are doing sir. Thanks :+1: 

Will Do My Best !!

**Comment by Keyikedalube** [comment]
> Please guide me what steps should I take if I want to get in GSOC with this organisation ??

@DhairyaBahl I opened issues that are too easy to solve so new contributors like you can get familiar with synfig codebase.

GSOC will be more problem solving rather than minor fixes here and there like mine. I would suggest you to take up issues that need creative problem solving. That way I guarantee you'll be ahead of the beginners curve :)

> Should I keep contributing to these issues ? OR should i work on some projects for the time being ??

Up to you :)

*Me personally I do my own gtkmm project first. I do my learning there, apply documentation/tutorials knowledge I studied, and then come back here to solve issues that I'm familiar with ;)*

**Pull request merged**